src/main/java/com/coveo/spillway/AddAndGetRequest.java

```diff
@@ -79,13 +97,68 @@ public Builder withEventTimestamp(Instant val) {
       return this;
     }
 
-    public Builder withIncrementBy(int val) {
-      incrementBy = val;
+    public Builder withCost(int val) {
+      cost = val;
       return this;
     }
 
     public AddAndGetRequest build() {
       return new AddAndGetRequest(this);
     }
   }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+
+    AddAndGetRequest that = (AddAndGetRequest) o;
+
+    if (cost != that.cost) return false;
+    if (resource != null ? !resource.equals(that.resource) : that.resource != null) return false;
```

Comment:
Ça fait longtemps que j'ai codé - so...

Une chose que je trouve plus difficile à lire, c'est le fait qu'il y a plusieurs return statements. Mais, ça, je peux vivre avec.

On pourrait faire une fonction du style.

if (different(resource, that.resource)) return false;

Au lieu de répéter la même forme pour chaque item?